src/languages/reasonml.js

```diff
@@ -6,301 +6,63 @@ Author: Gidi Meir Morris <oss@gidi.io>
-        contains: [
-          {
-            className: 'params',
-            variants: [
-              { begin: RE_IDENT },
-              { begin: RE_PARAM },
-              { begin: /\(\s*\)/ }
-            ]
-          }
-        ]
-      },
-      {
-        begin: '\\s\\(\\.?[^;\\|]*\\)\\s*=>',
-        end: '\\s=>',
-        returnBegin: true,
-        relevance: 0,
-        contains: [
-          {
-            className: 'params',
-            relevance: 0,
-            variants: [ PARAMS_MODE ]
-          }
-        ]
-      },
-      { begin: '\\(\\.\\s' + RE_IDENT + '\\)\\s*=>' }
-    ]
-  };
-  MODULE_ACCESS_CONTENTS.push(FUNCTION_BLOCK_MODE);
-
-  const CONSTRUCTOR_MODE = {
-    className: 'constructor',
-    begin: RE_MODULE_IDENT + '\\(',
-    end: '\\)',
-    illegal: '\\n',
-    keywords: KEYWORDS,
-    contains: [
-      hljs.QUOTE_STRING_MODE,
-      OPERATOR_MODE,
-      {
-        className: 'params',
-        begin: '\\b' + RE_IDENT
-      }
-    ]
-  };
-
-  const PATTERN_MATCH_BLOCK_MODE = {
-    className: 'pattern-match',
-    begin: '\\|',
-    returnBegin: true,
-    keywords: KEYWORDS,
-    end: '=>',
-    relevance: 0,
-    contains: [
-      CONSTRUCTOR_MODE,
-      OPERATOR_MODE,
-      {
-        relevance: 0,
-        className: 'constructor',
-        begin: RE_MODULE_IDENT
-      }
-    ]
-  };
-
-  const MODULE_ACCESS_MODE = {
-    className: 'module-access',
-    keywords: KEYWORDS,
-    returnBegin: true,
-    variants: [
-      { begin: "\\b(" + RE_MODULE_IDENT + "\\.)+" + RE_IDENT },
-      {
-        begin: "\\b(" + RE_MODULE_IDENT + "\\.)+\\(",
-        end: "\\)",
-        returnBegin: true,
-        contains: [
-          FUNCTION_BLOCK_MODE,
-          {
-            begin: '\\(',
-            end: '\\)',
-            relevance: 0,
-            skip: true
-          }
-        ].concat(MODULE_ACCESS_CONTENTS)
-      },
-      {
-        begin: "\\b(" + RE_MODULE_IDENT + "\\.)+\\{",
-        end: /\}/
-      }
-    ],
-    contains: MODULE_ACCESS_CONTENTS
-  };
-
-  PARAMS_CONTENTS.push(MODULE_ACCESS_MODE);
-
   return {
     name: 'ReasonML',
     aliases: [ 're' ],
-    keywords: KEYWORDS,
+    keywords: {
+      $pattern: '[a-z_]\\w*!?',
+      keyword:
+        'and as asr assert begin class constraint do done downto else end esfun '
+        + 'exception external for fun function functor if in include inherit '
+        + 'initializer land lazy let lor lsl lsr lxor mod module mutable new '
+        + 'nonrec object of open or pri pub rec sig struct switch then to try '
+        + 'type val virtual when while with',
+      built_in:
+        'array bool bytes char exn|5 float int int32 int64 list lazy_t|5 nativeint|5 ref string unit ',
+      literal:
+        'true false'
+    },
     illegal: '(:-|:=|\\$\\{|\\+=)',
     contains: [
-      hljs.COMMENT('/\\*', '\\*/', { illegal: '^(#,\\/\\/)' }),
-      {
-        className: 'character',
-        begin: '\'(\\\\[^\']+|[^\'])\'',
```

Comment:
Please use real literal regex whenever you can, not double-escaped strings, which makes it much harder to read the actual regex.  I realize that might have been how it was before, but if we're pretty much starting over we need to follow the conventions we use for our more modern grammars - which is to use real regex whenever possible.

Reply:
https://github.com/highlightjs/highlight.js/pull/3795/commits/ec0846191fbfeb48f1c57877fbd1e8fd57b3aab0